utils/subprocess/supervisor/interface.go

```diff
@@ -0,0 +1,9 @@
+package supervisor
+
+import "context"
+
+//go:generate mockgen -destination=../../mocks/mock_$GOPACKAGE.go -package=mocks github.com/ARM-software/golang-utils/utils/subprocess/$GOPACKAGE ISupervisor
+
+type ISupervisor interface {
+	Run(ctx context.Context) error
```

Comment:
```suggestion
       // Run will run the supervisor and execute any of the command hooks. If it receives a halting error or the context is cancelled then it will exit
	Run(ctx context.Context) error
```